### PR TITLE
Refactor code for Magento Coding Standards

### PR DESCRIPTION
**Pull request by mstrzyzewski**
-Modify classes descriptions
-Add missing log in the empty catch statement
-Add new Exceptions for placing reservations and processing orders
-Delete unused Exceptions
-Replace logic for API request with Guzzle

### CODE REVIEW

File: Model/OrderImporter/AllegroReservation.php [hunk 1]
```diff
@@ -111,9 +111,10 @@ public function placeReservation(CheckoutFormInterface $checkoutForm): void
...
                 $this->getSalesChannel(),
-                $this->getSalesEvent(self::ALLEGRO_EVENT_RESERVATION_PLACED, $checkoutForm->getId()));
+                $this->getSalesEvent(self::ALLEGRO_EVENT_RESERVATION_PLACED, $checkoutForm->getId())
+            );
         } catch (\Exception $e) {
-            throw new \Exception(
+            throw new ReservationPlacingException(
                 "Error while placing reservation for order with id [{$checkoutForm->getId()}]",
                 0,
                 $e
```
Comment (tmotyl) [inline Model/OrderImporter/AllegroReservation.php:120]: 0 is used as code? 
The code should be unique, to help find a place from where exception was thrown

File: Model/ResourceModel/Reservation/Collection/Grid.php [hunk 2]
```diff
@@ -32,10 +31,11 @@ public function __construct(
         FetchStrategy $fetchStrategy,
         EventManager $eventManager,
         $mainTable = 'allegro_reservations',
-        $resourceModel = Reservation::class,
+        $resourceModel = null,
         $identifierName = null,
         $connectionName = null
     ) {
+        $resourceModel = Reservation::class;
```
Comment (tmotyl) [inline Model/ResourceModel/Reservation/Collection/Grid.php:38]: Is this intentional? This change override whatever you pass as resourcemodel argument.

File: Model/ResourceModel/Sale/AfterSaleServices.php [hunk 3]
```diff
@@ -38,7 +44,8 @@ public function getReturnPoliciesList()
      */
     public function getWarrantiesList()
     {
-        $result = $this->cachedRequestGet('/after-sales-service-conditions/warranties?seller.id=' . $this->getCurrentUserId());
+        $result = $this
+            ->cachedRequestGet('/after-sales-service-conditions/warranties?seller.id=' . $this->getCurrentUserId());
```
Comment (tmotyl) [inline Model/ResourceModel/Sale/AfterSaleServices.php:48]: This looks strange

File: Plugin/InventorySourceItemSaveMultiple.php [hunk 4]
```diff
@@ -7,10 +7,7 @@
 use Magento\Framework\App\ResourceConnection;
 use Magento\InventoryApi\Api\Data\SourceItemInterface;
 
-/**
- * Class InventorySourceItemSaveMultiple
- * @package Macopedia\Allegro\Plugin
- */
+/** Puts message in a queue after save multiple action */
```
Comment (tmotyl) [inline Plugin/InventorySourceItemSaveMultiple.php:10]: Make it multiline

File: view/adminhtml/templates/origin_order_data.phtml [hunk 5]
```diff
@@ -4,11 +4,14 @@
  */
 ?>
 <?php if ($block->getExternalId()): ?>
-    <div class="admin__page-section-title">
-        <span class="title">
+    <div class="admin__page-section-title"> <span class="title">
```
Comment (tmotyl) [inline view/adminhtml/templates/origin_order_data.phtml:7]: It was more redeable when span was in the new line

File: Plugin/InventorySourceItemSaveMultiple.php (hunk 4 above)
Comment (mstrzyzewski) [reply Plugin/InventorySourceItemSaveMultiple.php]: Done

File: view/adminhtml/templates/origin_order_data.phtml (hunk 5 above)
Comment (mstrzyzewski) [reply view/adminhtml/templates/origin_order_data.phtml]: Done

File: Model/ResourceModel/Sale/AfterSaleServices.php (hunk 3 above)
Comment (mstrzyzewski) [reply Model/ResourceModel/Sale/AfterSaleServices.php]: Done

File: Model/ResourceModel/Reservation/Collection/Grid.php (hunk 2 above)
Comment (mstrzyzewski) [reply Model/ResourceModel/Reservation/Collection/Grid.php:38]: I did it to resolve "Possible useless method overriding detected" warning. What's the best solution for this warning?

File: Model/OrderImporter/AllegroReservation.php (hunk 1 above)
Comment (mstrzyzewski) [reply Model/OrderImporter/AllegroReservation.php:120]: If this code should be unique how can I check if code that I will make up is not used by other Exception? Should I also declare this code in my Exception class as a constant variable?

Comment (tmotyl) [reply Model/OrderImporter/AllegroReservation.php:120]: nop, just use current timestamp as a code

File: Model/ResourceModel/Reservation/Collection/Grid.php (hunk 2 above)
Comment (tmotyl) [reply Model/ResourceModel/Reservation/Collection/Grid.php:38]: did you check what is the rule behind "Possible useless method overriding detected" warning?

Comment (mstrzyzewski) [reply Model/ResourceModel/Reservation/Collection/Grid.php]: Done

File: Model/OrderImporter/AllegroReservation.php (hunk 1 above)
Comment (mstrzyzewski) [reply Model/OrderImporter/AllegroReservation.php:120]: Done

File: Controller/Adminhtml/Offer/Save.php [hunk 6]
```diff
@@ -67,15 +67,21 @@ public function execute()
                     $this->productRepository->save($product);
                 } catch (CouldNotSaveException $e) {
                     $this->messageManager->addWarningMessage(
-                        __('Could not assign offer id to product. Please update product data with proper offer ID manually')
+                        __('Could not assign offer id to product. Please update product data with proper offer ID manually')//phpcs:ignore
```
Comment (tmotyl) [inline Controller/Adminhtml/Offer/Save.php:70]: what was phpcs complaining about here?

File: Controller/Adminhtml/System/Authenticate.php [hunk 7]
```diff
@@ -62,7 +62,11 @@ public function execute()
                 $this->getMessageManager()->addErrorMessage(__('Something went wrong while authorization in Allegro.'));
                 $this->getMessageManager()->addExceptionMessage($exception);
             } catch (\Exception $exception) {
-                $this->getMessageManager()->addErrorMessage(__('Something went wrong while authorization in Allegro. Please check credentials and try again'));
+                $this
+                    ->getMessageManager()
```
Comment (tmotyl) [inline Controller/Adminhtml/System/Authenticate.php:66]: should be the same line with $this.

File: Model/Api/Client.php [hunk 8]
```diff
@@ -26,53 +32,68 @@ class Client
...
-                __(
-                    'Errors returned in received from Allegro API response: %1',
-                    implode(' ', $errors)
-                )
-            );
-        }
 
-        return $response;
+            return $response;
+        } catch (GuzzleException $e) {
```
Comment (tmotyl) [inline Model/Api/Client.php:94]: which line throws the GuzzleException ? the
```
$json = $this->sendHttpRequest($token, $request);
```
if yes, then we should wrap just it in the try-catch.

File: Model/OrderImporter/AllegroReservation.php (hunk 1 above)
Comment (tmotyl) [reply Model/OrderImporter/AllegroReservation.php:120]: it has to be static value (the time when you wrote the line of code) e.g. 1589537388. Making it dynamic would make it impossible to find the error by code, as every time the code would be different.

File: Model/OrderImporter/AllegroReservation.php [hunk 9]
```diff
@@ -137,9 +138,9 @@ public function compensateReservation(CheckoutFormInterface $checkoutForm): void
                 $this->getSalesEvent(self::ALLEGRO_EVENT_RESERVATION_COMPENSATED, $checkoutForm->getId())
             );
         } catch (\Exception $e) {
-            throw new \Exception(
+            throw new ReservationPlacingException(
                 "Error while compensating reservation for order with id [{$checkoutForm->getId()}]",
-                0,
+                time(),
```
Comment (tmotyl) [inline Model/OrderImporter/AllegroReservation.php:143]: same here

File: Model/OrderImporter/AllegroReservation.php [hunk 10]
```diff
@@ -172,7 +173,8 @@ private function getSalesChannel(): SalesChannelInterface
             $storeId = Store::DEFAULT_STORE_ID;
         }
 
-        $websiteCode = $this->storeManager->getWebsite($this->storeManager->getStore($storeId)->getWebsiteId())->getCode();
+        $websiteCode = $this->storeManager
```
Comment (tmotyl) [inline Model/OrderImporter/AllegroReservation.php:176]: getWebsite should be in the upper line

File: Model/OrderImporter/Processor.php [hunk 11]
```diff
@@ -167,8 +167,11 @@ private function addOrderWithErrorToTable(CheckoutFormInterface $checkoutForm, \
         try {
             $this->orderLogRepository->save($orderLog);
         } catch (CouldNotSaveException $e) {
-            throw new \Exception("Error while adding order with id [{$checkoutFormId}] to allegro_orders_with_errors table",
-                0, $e);
+            throw new OrderProcessingException(
+                "Error while adding order with id [{$checkoutFormId}] to allegro_orders_with_errors table",
+                time(),
```
Comment (tmotyl) [inline Model/OrderImporter/Processor.php:172]: same here

File: Model/OrderImporter/Processor.php [hunk 12]
```diff
@@ -183,7 +186,7 @@ private function tryCreateOrder(CheckoutFormInterface $checkoutForm): void
             $this->creator->execute($checkoutForm);
             $this->logger->info("Order with id [$checkoutFormId] has been successfully created");
         } catch (\Exception $e) {
-            throw new \Exception("Error while creating order with id [{$checkoutFormId}]", 0, $e);
+            throw new OrderProcessingException("Error while creating order with id [{$checkoutFormId}]", time(), $e);
```
Comment (tmotyl) [inline Model/OrderImporter/Processor.php:189]: and here

File: Model/OrderImporter/Processor.php [hunk 13]
```diff
@@ -197,8 +200,11 @@ private function removeErrorLogIfExist(CheckoutFormInterface $checkoutForm): voi
         try {
             $this->orderLogRepository->deleteByCheckoutFormId($checkoutFormId);
         } catch (CouldNotDeleteException $e) {
-            throw new \Exception("Error while deleting order with id [{$checkoutFormId}] from allegro_orders_with_errors table",
-                0, $e);
+            throw new OrderProcessingException(
+                "Error while deleting order with id [{$checkoutFormId}] from allegro_orders_with_errors table",
+                time(),
```
Comment (tmotyl) [inline Model/OrderImporter/Processor.php:205]: ^^

File: Model/ResourceModel/Reservation/Collection/Grid.php [hunk 14]
```diff
@@ -16,6 +15,7 @@ class Grid extends SearchResult
 {
     /**
      * Grid constructor.
+     * phpcs:disable Generic.CodeAnalysis.UselessOverridingMethod
```
Comment (tmotyl) [inline Model/ResourceModel/Reservation/Collection/Grid.php:18]: cheating!
fix the error instead of suppressing it.

File: Model/ResourceModel/Sale/AfterSaleServices.php [hunk 15]
```diff
@@ -14,7 +14,10 @@ class AfterSaleServices extends AbstractResource
      */
     public function getImpliedWarrantiesList()
     {
-        $result = $this->cachedRequestGet('/after-sales-service-conditions/implied-warranties?seller.id=' . $this->getCurrentUserId());
+        $result = $this
+            ->cachedRequestGet(
```
Comment (tmotyl) [inline Model/ResourceModel/Sale/AfterSaleServices.php:18]: cachedRequestGet should be in the same line as $this

File: Model/ResourceModel/Sale/AfterSaleServices.php [hunk 16]
```diff
@@ -26,7 +29,10 @@ public function getImpliedWarrantiesList()
      */
     public function getReturnPoliciesList()
     {
-        $result = $this->cachedRequestGet('/after-sales-service-conditions/return-policies?seller.id=' . $this->getCurrentUserId());
+        $result = $this
+            ->cachedRequestGet(
```
Comment (tmotyl) [inline Model/ResourceModel/Sale/AfterSaleServices.php:33]: same here

File: Model/ResourceModel/Sale/Images.php [hunk 17]
```diff
@@ -2,23 +2,52 @@
...
+        Logger $logger,
+        File $file
+    ) {
+        parent::__construct($scopeConfig, $client, $tokenProvider, $logger);
+        $this->file = $file;
+    }
+
     /**
-     * @param string $imageUrl
+     * @param $imagePath
```
Comment (tmotyl) [inline Model/ResourceModel/Sale/Images.php:37]: why did you remove the type?

File: Observer/ConfigCredentialsChangeObserver.php [hunk 18]
```diff
@@ -61,9 +61,9 @@ public function execute(Observer $observer)
             try {
                 $this->credentials->getToken();
                 $this->credentials->deleteToken();
-                $this->messageManager->addNoticeMessage(__('You have changed credentials to Allegro account. Your current connection has been lost and you have to connect with Allegro account again'));
+                $this->messageManager->addNoticeMessage(__('You have changed credentials to Allegro account. Your current connection has been lost and you have to connect with Allegro account again'));//phpcs:ignore
```
Comment (tmotyl) [inline Observer/ConfigCredentialsChangeObserver.php:64]: why suppressing the error?

File: view/adminhtml/templates/origin_order_data.phtml [hunk 19]
```diff
@@ -9,6 +9,10 @@
         <?= $block->escapeHtml(__('Origin of the order')) ?>
         </span>
     </div>
-    <p><strong><?= $block->escapeHtml(__('External service')) ?></strong>: <?= $block->escapeHtml($block->getOrderOrigin()) ?></p>
-    <p><strong><?= $block->escapeHtml(__('External ID')) ?></strong>: <?= $block->escapeHtml($block->getExternalId()) ?></p>
+    <p> <strong><?= $block->escapeHtml(__('External service')) ?></strong>
+        :
```
Comment (tmotyl) [inline view/adminhtml/templates/origin_order_data.phtml:13]: can't the colon be in the same line as "External service" ?

File: Model/ResourceModel/Sale/AfterSaleServices.php [hunk 20]
```diff
@@ -26,7 +29,10 @@ public function getImpliedWarrantiesList()
      */
     public function getReturnPoliciesList()
     {
-        $result = $this->cachedRequestGet('/after-sales-service-conditions/return-policies?seller.id=' . $this->getCurrentUserId());
+        $result =
+            $this->cachedRequestGet(
```
Comment (tmotyl) [inline Model/ResourceModel/Sale/AfterSaleServices.php:33]: `$result = $this->cachedRequestGet(`

File: Model/ResourceModel/Sale/AfterSaleServices.php [hunk 21]
```diff
@@ -38,7 +44,10 @@ public function getReturnPoliciesList()
      */
     public function getWarrantiesList()
     {
-        $result = $this->cachedRequestGet('/after-sales-service-conditions/warranties?seller.id=' . $this->getCurrentUserId());
+        $result =
+            $this->cachedRequestGet(
```
Comment (tmotyl) [inline Model/ResourceModel/Sale/AfterSaleServices.php:48]: here also

File: view/adminhtml/templates/origin_order_data.phtml [hunk 22]
```diff
@@ -9,6 +9,8 @@
         <?= $block->escapeHtml(__('Origin of the order')) ?>
         </span>
     </div>
-    <p><strong><?= $block->escapeHtml(__('External service')) ?></strong>: <?= $block->escapeHtml($block->getOrderOrigin()) ?></p>
-    <p><strong><?= $block->escapeHtml(__('External ID')) ?></strong>: <?= $block->escapeHtml($block->getExternalId()) ?></p>
+    <p> <strong><?= $block->escapeHtml(__('External service')) ?></strong> :
```
Comment (tmotyl) [inline view/adminhtml/templates/origin_order_data.phtml:12]: there should be no space before ":"
